src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/ContentItemsFieldType.cs

```diff
@@ -172,15 +172,18 @@ private async Task<IEnumerable<ContentItem>> Resolve(ResolveFieldContext context
             return contentQuery;
         }
 
-        private IQuery<ContentItem> PageQuery(IQuery<ContentItem> contentItemsQuery, ResolveFieldContext context)
+        private IQuery<ContentItem> PageQuery(IQuery<ContentItem> contentItemsQuery, ResolveFieldContext context, GraphQLContext graphQLContext)
         {
-            if (context.HasPopulatedArgument("first"))
-            {
-                var first = context.GetArgument<int>("first");
+            var first = context.GetArgument<int>("first");
 
-                contentItemsQuery = contentItemsQuery.Take(first);
+            if (first == 0)
+            {
+                var serviceProvider = graphQLContext.ServiceProvider;
+                first = serviceProvider.GetService<IOptions<GraphQLSettings>>().Value.DefaultNumberOfResults;
```

Comment:
This class should be resolved too, so you could just read it once in a local variable instead of resolving the service on every query